app/components/publications/publication/translations/translation-request-modal.js

```diff
@@ -2,59 +2,91 @@ import Component from '@glimmer/component';
 import { action } from '@ember/object';
 import { inject as service } from '@ember/service';
 import { tracked } from '@glimmer/tracking';
-import { task } from 'ember-concurrency-decorators';
+import { task, dropTask } from 'ember-concurrency-decorators';
 import { translationRequestEmail } from 'frontend-kaleidos/utils/publication-email';
-import { add } from 'ember-math-helpers/helpers/add';
+import { Validator, ValidatorSet } from 'frontend-kaleidos/utils/validators';
+import { isPresent } from '@ember/utils';
 
 export default class PublicationsTranslationRequestModalComponent extends Component {
   /**
-   * @argument onSave: should take arguments (selectedPieces)
+   * @argument dueDate
+   * @argument publicationFlow
+   * @argument onSave
    * @argument onCancel
    */
   @service store;
 
-  @tracked translationDueDate = this.args.translationSubcase.dueDate ? this.args.translationSubcase.dueDate : new Date();
+  @tracked uploadedPieces = [];
+  @tracked pagesAmount;
+  @tracked wordsAmount;
+  @tracked translationDueDate = this.args.dueDate
+    ? this.args.dueDate
+    : new Date();
   @tracked subject;
   @tracked message;
+  validators;
 
   constructor() {
     super(...arguments);
+    this.initValidators();
     this.setEmailFields.perform();
   }
 
-  get totalPages() {
-    const pages = this.args.attachments.mapBy('pages').filter((page) => page >= 0);
-    return pages.length ? add(pages) : 0;
+  get isCancelDisabled() {
+    return this.cancel.isRunning || this.save.isRunning;
   }
 
-  get totalWords() {
-    const words = this.args.attachments.mapBy('words').filter((word) => word >= 0);
-    return words.length ? add(words) : 0;
+  get isSaveDisabled() {
+    return (
+      this.uploadedPieces.length === 0 ||
+      !this.validators.areValid ||
+      this.cancel.isRunning
+    );
   }
 
-  get totalDocuments() {
-    return this.args.attachments.length;
+  @task
+  *save() {
+    yield this.args.onSave({
+      uploadedPieces: this.uploadedPieces,
+      translationDueDate: this.translationDueDate,
+      subject: this.subject,
+      message: this.message,
+    });
   }
 
-  get isSaveDisabled() {
-    return this.translationDueDate === null
-      || this.subject === null
-      || this.message === null;
+  @dropTask
+  *cancel() {
+    // necessary because close-button is not disabled when saving
+    if (this.save.isRunning) {
+      return;
+    }
+    yield Promise.all(this.uploadedPieces.map((piece) => this.deleteUploadedPiece.perform(piece)));
+    this.args.onCancel();
+  }
+
+  @dropTask
+  *deleteUploadedPiece(piece) {
+    const file = yield piece.file;
+    const documentContainer = yield piece.documentContainer;
+    yield Promise.all([
+      file.destroyRecord(),
+      documentContainer.destroyRecord(),
+    ]);
+    this.uploadedPieces.removeObject(piece);
```

Comment:
1- Hier wordt piece zelf niet verwijderd, model is nog niet gesaved maar een cleanup is snel gebeurd.

2- Dit is een lastige qua code inconsistentie.
Op 6 andere plaatsen in onze code doen we dit process (file, container en piece verwijderen)
Op 3 van die plaatsen (agenda/agendaitem/subcase is de code 100% duplicate (mogelijkheid tot util of service)
De andere 3 plaatsen (+ deze dus 4) zitten in publicatie en ze zijn allemaal anders.
Er wordt meer met promises gewerkt, maar telkens op een andere manier. Die droptask erbij maakt het hier nog verwarrender.
Ook is er de fileService, die ook gelijkaardig dingen kan opkuisen (maar heeft nood aan een likje "task" verf.)

**Moet niet in deze PR gelijk getrokken worden,** ik merk het gewoon op.
een voorbeeld van zo'n duplicate
https://github.com/kanselarij-vlaanderen/frontend-kaleidos/blob/feature/KAS-3106_Vertalingen_herwerken/app/pods/agenda/agendaitems/agendaitem/documents/controller.js#L126